definition_validator/validator.cc

```diff
@@ -537,7 +525,10 @@ void validateOverriding(const core::Context ctx, const ast::ExpressionPtr &tree,
     // we don't raise override errors if the method implements an abstract method, which means we need to know ahead of
     // time whether any parent methods are abstract
     auto anyIsInterface = absl::c_any_of(overriddenMethods, [&](auto &m) { return m.data(ctx)->flags.isAbstract; });
+    auto methodHasSig = method.data(ctx)->hasSig() || method.data(ctx)->flags.isOverloaded;
     for (const auto &overriddenMethod : overriddenMethods) {
+        auto overriddenMethodHasSig =
+            overriddenMethod.data(ctx)->hasSig() || overriddenMethod.data(ctx)->flags.isOverloaded;
```

Comment:
This is an interesting consequence of finding the method symbol, not the first signature: the symbol that we find no longer returns `true` for `hasSig`. However, knowing that `isOverloaded` is true implies that there is a signature somewhere, and as we don't use the signature in this context, knowing that it exists is enough.

It might be reasonable to modify `hasSig` to also take into account the overload status of the method, but we use it in multiple places to guard access to the actual method type, so this would be a little odd without coupling it with a lookup of the first overload.